### DISCUSSION TITLE
Why should wxt continue to support Manifest V2 despite its deprecation and phase-out within a year?

### DISCUSSION THREAD

**Discussion by ayub-kokabi** [wxt-dev/wxt#800] [General]
[The deprecation of Manifest V2 has already begun, and it will be completely phased out within a year](https://blog.chromium.org/2024/05/manifest-v2-phase-out-begins.html) and wxt is a new project, why should it continue to support Manifest V2? 

I think doing so not only increases WXT's development cost but also makes maintaining documentation for both Manifest V2 and V3 challenging. Additionally, a portion of development effort will likely be spent on discussing Manifest V2 compatibility issues. Furthermore, removing Manifest V2 from the codebase in the future could be difficult, expensive, and prone to errors as the project grows larger. 

So why not ditch V2 now and focus on the future?

**Comment by aiktb** [comment]
Firefox is committed to retaining MV2, and Firefox has very limited support for MV3 at the moment - some Firefox MV2 extensions break once you change the build target to MV3 (even if they work fine in Chrome MV3).

Only the `wxt build -b firefox` command has a default build target of MV2, which shouldn't affect much if you only care about Chrome.

And WXT is considering removing `webextension-polyfill`, which is probably what you want: https://github.com/wxt-dev/wxt/issues/784

**Reply by ayub-kokabi** [reply]
Thank you. 🙏
However, [Firefox only holds 3% of the browser market share](https://gs.statcounter.com/browser-market-share#monthly-202306-202406-bar) and it's not really worth paying attention to their decisions. Sooner or later, they'll have to make choices based on what the majority of the market dictates.

**Reply by aklinker1** [reply]
I'm one of those 3%, so ditching Firefox is not an option 🤷. Safari also has not dropped MV2 support, and AFAIK, they haven't announced anything either.

Surprisingly, MV2 support is really easy. The main difference is generating the manifest a little differently. For dev mode, I think there's 1 or 2 if-statements? So not as much complexity as you would think. You are right about the cost of answering questions though lol.

Related:
- https://github.com/wxt-dev/wxt/issues/230

**Reply by ruihildt** [reply]
@aklinker1 This explains it why Firefox is so well supported. Very much appreciated. ❤️
